### PR TITLE
Adds parsing of generic package warnings

### PR DESCRIPTION
**Pull request by mathsguy42**
Added a single function to handle all package warnings, either one-line or multi-line, and finding the correct line number if any.

Instead of crafting a regex for each type of error we can take advantage of the fact that LaTeX inserts a blank line after each warning, regardless of the length of the warning message. I also added the code to look for the line number in each line of multi-line warnings.

For the case of biber warnings this makes the parsing a lot faster, because it stops as soon as it finds a blank line, instead of reading the whole log file.

### CODE REVIEW

**Comment by jpallen** [comment]
Awesome! Please can you add a new quick test case (see tests/) to check that the code is working for packages like geometry that it previously failed to pick up.

**Comment by mathsguy42** [comment]
Upon further investigation I had to update the parser, some packages insert a prefix (<packagename>) in multi-line comments and some don't, so it was necessary to identify the package name and delete that prefix to keep the resulting parsed message clean. Besides that, the main changes are:

1. Modified the parser to include the string "Package (<packageame>) Warning:", this makes it easier to figure out what is causing the warning, specially when the message does not include the line number.

2. Added tests for the `geometry` package, since the uncaught warnings there prompted us to update the parser.

3. Added tests for the `caption` package. Warnings from this one are hard to parse because the second line has a *(caption)* prefix and also includes the source-file line number, and the third line don't have blank spaces at the beginning.

**Comment by jpallen** [comment]
Great, thanks :)
